qiskit/transpiler/passes/optimization/optimize_1q_decomposition.py

```diff
@@ -70,8 +70,13 @@ def run(self, dag):
         runs = dag.collect_runs(self.euler_basis_names[self.basis])
         runs = _split_runs_on_parameters(runs)
         for run in runs:
-            # Don't try to optimize a single 1q gate
             if len(run) <= 1:
+                params = run[0].op.params
+                # Remove single identity gates
+                if run[0].op.name in self.euler_basis_names[self.basis] and len(
+                        params) > 0 and all(param == 0 for param in params):
```

Comment:
The `all(param == 0 ...)` check may miss some cases (e.g. for R gates). Can we check something like `np.array_equal(op.to_matrix(), np.eye(2))` instead?

Reply:
Yeah, that's a good idea. Done in: https://github.com/Qiskit/qiskit-terra/pull/5292/commits/0fdfcce433ac1a800938344e0fb6c7a3335cb7d3